joomla/joomla-cms
# [4.2] Fix MailTemplate array regression from 4.2.7

Pull Request for Issue #39774.

### Summary of Changes
PR #38305 changed the logic for array values in mail templates. That PR "fixed" the output of plain arrays, dismissing that core actually uses arrays in a _different_ way, by using nested template tags.

One feature where nested tags are used are the user information emails, and thus they are broken since 4.2.7.

This PR restores the old behavior, but only if the new one is not used. As described by @ReLater in 39774, this fixes the mails not working (on my sites, with default templates) - and it also doesn't break the use-case of the changes in 4.2.7.

#### Additional remarks on #38305

Note: I find the changes in #38305 confusing as is breaks the current semantics of templates: beforehand, only text in curly braces was treated as "key", and now also text outside of them. imho, `{com_fields}{xyz}{/com_fields}` would have been better than the syntax `{com_fields}xyz{/com_fields}` as implemented by #38305...

I suppose the decision on how to put forward the mail templating "language" is not on me, but this PR at least unbreaks old functionality and does not introduce b/c problems to 4.2.7.


### Testing Instructions
* Install joomla 4.2.7
* Make sure action logs for user events are enabled and configure working system email sending
* In the user settings for a user with a valid email address (you can receive mail on), enable user action information emails for the user component.
* Log out and log back in
* Look at the emails you should have received


### Actual result BEFORE applying this Pull Request
* You get information emails about user activity, but they only contain the table header and no specific contents,


### Expected result AFTER applying this Pull Request
* You get information emails that also contain information about the user action (like <=4.2.6)


### Link to documentations
Please select:
